streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -428,6 +441,9 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
     private val submitJobThreadPool = ExecutionContext.fromExecutorService(
       ThreadUtils.newDaemonCachedThreadPool("submit-job-thead-pool"))
 
+    private val walBatchingThreadPool = ExecutionContext.fromExecutorService(
```

Comment:
It's better to shutdown this thread pool in `onStop`. If `addBlockAsync` is very slow, we may exhaust threads because the number of `block` could be a big number. How about making `addBlockAsync` become real asynchronous. The `addBlockAsync` would become

```
def addBlockAsync(event): Future[...] = {
   put the event to queue with a Promise
   promise.future.mapTo { v => 
       do some work after writing 
   }
}
```